### DISCUSSION TITLE
issues building gnuplot 5.4.0.1 - what tools should I be using?

### DISCUSSION THREAD

**Discussion by crenelle** [Homebrew/discussions#87] [Everyday usage]
I'm not sure where I should post this.

The current gnuplot built using brew doesn't successfully execute many of the .dem files in the demo folder. If I hand gnuplot a script longer than a few lines I'll see a "Bus error: 10", and simple script files with embedded data blocks seemingly get confused that the lines of data are command lines it can't recognize. I suspect I'm using a toolset that's somewhat incompatible with the brew gnuplot build.

For comparison, I am able to build a macports gnuplot build, that does execute the .dem files in the demo folder. I prefer using the brew build because the qt terminal seems less quirky.

brew seems to upgrade itself and all the installed tools without problems.

**Comment by SMillerDev** [comment]
What commands are you running? What does work and what doesn't?

**Reply by crenelle** [reply]
I've been using gnuplot built using brew by launching from an app I'm building. gnuplot is working pretty well from that aspect.

When I try to run gnuplot using Terminal under macOS and just run some of the demo scripts, things go awry. For example:

bentley$ gnuplot jitter.dem

set term aqua
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

Bus error: 10

bentley$ gnuplot palette+alpha.dem

set term aqua
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

Segmentation fault: 11

I see the problem with "set term aqua" all the time; it must be built into gnuplot because I'm not doing it, nor are the .dem files.

Most of the demo files don't appear to work. One that does is finance.dem.

-Mike

**Reply by jonchang** [reply]
I don't think our build supports aquaterm (which I believe is what `set term aqua` is doing)? We use the qt backend for gnuplot.

**Comment by crenelle** [comment]
I don't think it does either. I'm wondering why the build starts runtime by executing set term aqua. All my scripts do a set term qt.

**Reply by gromgit** [reply]
Check your `/usr/local/share/gnuplot/5.4/gnuplotrc` and `~/.gnuplot`. Sounds like one or the other has a `set term aqua` in it.

**Reply by crenelle** [reply]
Did not know those existed. Ok, taking a look, I don't see set term aqua in the indicated gnuplotrc file or ~/.gnuplot (I only found a .gnuplot_history). The three gnuplotrc files I found (each related to a separate repo or build) contained nothing but commented out suggestions for things that should be done in the rc file.

**Reply by jonchang** [reply]
I looked at the man page for gnuplot and it looks like there's a GNUTERM environment variable that might be causing this.

**Comment by crenelle** [comment]
I don't seem to have any well hidden gnuplot setup in relevant shell setups. I have a .profile in ~ that doesn't do anything. I checked out the source for gnuplot-main and searched for "set term aqua". The phrase does show up in comments.
```
bentley$ gnuplot

	G N U P L O T
	Version 5.4 patchlevel 0    last modified 2020-07-13 

	Copyright (C) 1986-1993, 1998, 2004, 2007-2020
	Thomas Williams, Colin Kelley and many others

	gnuplot home:     http://www.gnuplot.info
	faq, bugs, etc:   type "help FAQ"
	immediate help:   type "help"  (plot window: hit 'h')

Terminal type is now 'unknown'

set term aqua
         ^
unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

gnuplot> show loadpath

	loadpath is empty
	gnuplotrc is read from /usr/local/Cellar/gnuplot/5.4.0_1/share/gnuplot/5.4

gnuplot> quit
```
```
bentley$ cd /usr/local/Cellar/gnuplot/5.4.0_1/share/gnuplot/5.4
bentley$ ls
PostScript		colors_mono.gp		gnuplot.gih		js			qt
colors_default.gp	colors_podo.gp		gnuplotrc		lua
bentley$ more gnuplotrc
###
### Gnuplot version 5.0 initialization file
### This file is loaded by gnuplot at the start of each run.
### It is provided as a template, with all commands commented out.
### Uncomment and customize lines for local use.
### Any commands placed here will affect all users.
### To customize gnuplot's initial state for an individual user,
### place commands in a private file ~/.gnuplot instead.

###
### Language initialization
###
# set locale
# set encoding locale

###
### Default line colors and repeat cycle
###
# set linetype 1 lc rgb "dark-violet" lw 1
# set linetype 2 lc rgb "#009e73" lw 1
# set linetype 3 lc rgb "#56b4e9" lw 1
# set linetype 4 lc rgb "#e69f00" lw 1
# set linetype 5 lc rgb "#f0e442" lw 1
# set linetype 6 lc rgb "#0072b2" lw 1
# set linetype 7 lc rgb "#e51e10" lw 1
# set linetype 8 lc rgb "black"   lw 1
# set linetype cycle 8

###
### Initialize the default loadpath for shared gnuplot scripts and data.
### Please confirm that this path is correct before uncommented the line below.
###
# set loadpath "/usr/local/share/gnuplot/4.7/demo"

###
### Some commonly used functions that are not built in
###
# sinc(x) = sin(x)/x
# rgb(r,g,b) = sprintf("#%06x",256.*256.*255.*r+256.*255.*g+255.*b)
# hsv(h,s,v) = sprintf("#%06x",hsv2rgb(h,s,v))

###
### Other preferences
###
# set clip two
```

**Reply by gromgit** [reply]
What does `echo $GNUTERM` output from your command line? My own tests seem to agree with @jonchang's supposition that you have that environment variable set to `aqua`:
```
$ gnuplot

	G N U P L O T
	Version 5.4 patchlevel 0    last modified 2020-07-13 

	Copyright (C) 1986-1993, 1998, 2004, 2007-2020
	Thomas Williams, Colin Kelley and many others

	gnuplot home:     http://www.gnuplot.info
	faq, bugs, etc:   type "help FAQ"
	immediate help:   type "help"  (plot window: hit 'h')

Terminal type is now 'qt'
gnuplot> 

$ GNUTERM=aqua gnuplot

	G N U P L O T
	Version 5.4 patchlevel 0    last modified 2020-07-13 

	Copyright (C) 1986-1993, 1998, 2004, 2007-2020
	Thomas Williams, Colin Kelley and many others

	gnuplot home:     http://www.gnuplot.info
	faq, bugs, etc:   type "help FAQ"
	immediate help:   type "help"  (plot window: hit 'h')

Terminal type is now 'unknown'

set term aqua
         ^
unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

gnuplot> 
```

**Reply by crenelle** [reply]
That was it, I (completely) overlooked the entry in the .profile. Accessibility:zoom for the out of the money.

-m
